homeassistant/components/blebox/binary_sensor.py

```diff
@@ -0,0 +1,54 @@
+"""BleBox binary sensor entities."""
+
+from blebox_uniapi.binary_sensor import BinarySensor as BinarySensorFeature
+
+from homeassistant.components.binary_sensor import (
+    BinarySensorDeviceClass,
+    BinarySensorEntity,
+    BinarySensorEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import BleBoxEntity, create_blebox_entities
+
+BINARY_SENSOR_TYPES = (
+    BinarySensorEntityDescription(
+        key="moisture",
+        device_class=BinarySensorDeviceClass.MOISTURE,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up a BleBox entry."""
+
+    create_blebox_entities(
+        hass,
+        config_entry,
+        async_add_entities,
+        BleBoxBinarySensorEntity,
+        "binary_sensors",
+        BINARY_SENSOR_TYPES,
+    )
```

Comment:
You misunderstood me. I was refering to `BleBoxBinarySensorEntity.__init__()` not refering to `__init__` module.

I think that you should NOT reference `create_blebox_entities`, and instead move the logic here.

```suggestion
    product: Box = hass.data[DOMAIN][config_entry.entry_id][PRODUCT]
    entities: list[BleBoxBinarySensorEntity] = []

    if "binary_sensors" in product.features:
        for feature in product.features["binary_sensors"]:
            for description in BINARY_SENSOR_TYPES:
                if description.key == feature.device_class:
                    entities.append(entity_klass(feature, description))
                    break
    async_add_entities(entities, True)
```